tests/patch/test_torch_export_patches.py

```diff
@@ -0,0 +1,247 @@
+            te = expected.astype(int) if expected.dtype == np.bool_ else expected
+            tv = value.astype(int) if value.dtype == np.bool_ else value
+            rows = [
+                f"{msg}\n{e}" if msg else str(e),
+                f"expected max value={expected_max}",
+                f"expected computed value={expected_value}\n",
+                f"ratio={te / tv}\ndiff={te - tv}",
+            ]
+            raise AssertionError("\n".join(rows))
+
+    # @skipif_ci_windows("not working on Windows")
+    def test_pytree_flatten_mamba_cache(self):
+        import torch
+        import torch.utils._pytree as py_pytree
+
+        try:
+            from transformers.models.mamba.modeling_mamba import MambaCache
+        except ImportError:
+            from transformers.cache_utils import MambaCache
+
+        class Config:
+            def __init__(self):
+                self.intermediate_size = 8
+                self.state_size = 16
+                self.conv_kernel = 32
+                self.num_hidden_layers = 64
+                self.dtype = torch.float16
+
+        cache = MambaCache(Config(), max_batch_size=1, device="cpu")
+
+        with torch_export_patches(verbose=1):
+            values, spec = py_pytree.tree_flatten(cache)
+            cache2 = py_pytree.tree_unflatten(values, spec)
+            self.assertEqual(cache.max_batch_size, cache2.max_batch_size)
+            self.assertEqual(cache.intermediate_size, cache2.intermediate_size)
+            self.assertEqual(cache.ssm_state_size, cache2.ssm_state_size)
+            self.assertEqual(cache.conv_kernel_size, cache2.conv_kernel_size)
+            self.assertEqualArrayAny(cache.conv_states, cache2.conv_states)
+            self.assertEqualArrayAny(cache.ssm_states, cache2.ssm_states)
+
+    # @skipif_ci_windows("not working on Windows")
+    def test_exportable_mamba_cache(self):
+        import torch
+        from transformers.models.mamba.modeling_mamba import MambaCache
+
+        class Config:
+            def __init__(self):
+                self.intermediate_size = 8
+                self.state_size = 16
+                self.conv_kernel = 32
+                self.num_hidden_layers = 64
+                self.dtype = torch.float16
+
+        class Model(torch.nn.Module):
+            def forward(self, x: torch.Tensor, cache: MambaCache):
+                x1 = cache.ssm_states[0] + x
+                x2 = cache.conv_states[0][:, :, ::2] + x1
+                return x2
+
+        cache = MambaCache(Config(), max_batch_size=1, device="cpu")
+        x = torch.ones(2, 8, 16).to(torch.float16)
+        model = Model()
+        model(x, cache)
+
+        with torch_export_patches(verbose=1, patch_transformers=True):
+            cache = MambaCache(Config(), max_batch_size=1, device="cpu")
+            torch.export.export(Model(), (x, cache))
+
+    # @skipif_ci_windows("not working on Windows")
+    def test_exportable_mamba_cache_dynamic(self):
```

Comment:
This test weirdly fails when it's run alone, but passes when it's run with others.

Reply:
You can skip it. We'll put MambaCache back in the patches when needed. transformers has started some refactorization as well for this one. At least, they are moving its location in the package.